File: composeApp/src/main/kotlin/com/po4yka/trailglass/ui/screens/timeline/TimelineCards.kt
package com.po4yka.trailglass.ui.screens.timeline

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.DirectionsWalk
import androidx.compose.material.icons.automirrored.filled.HelpOutline
import androidx.compose.material.icons.automirrored.filled.StickyNote2
import androidx.compose.material.icons.filled.CalendarMonth
import androidx.compose.material.icons.filled.CalendarToday
import androidx.compose.material.icons.filled.DateRange
import androidx.compose.material.icons.filled.Place
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.filled.Straighten
import androidx.compose.material3.AssistChip
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.po4yka.trailglass.domain.model.PlaceCategory
import com.po4yka.trailglass.feature.timeline.GetTimelineUseCase

@Composable
fun DayMarkerCard(
    text: String,
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors =
            CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.surfaceVariant
            )
    ) {
        Row(
            modifier = Modifier.padding(12.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            Icon(icon, contentDescription = null, tint = MaterialTheme.colorScheme.primary)
            Text(
                text,
                style = MaterialTheme.typography.bodyMedium,
                fontWeight = FontWeight.Medium
            )
        }
    }
}

@Composable
fun EnhancedVisitCard(
    visit: com.po4yka.trailglass.domain.model.PlaceVisit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors =
            CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.primaryContainer
            )
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            // Header
            Row(
                verticalAlignment = Alignment.Top,
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                Icon(
                    getCategoryIcon(visit.category),
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onPrimaryContainer,
                    modifier = Modifier.size(28.dp)
                )

                Column(modifier = Modifier.weight(1f)) {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text(
                            text = visit.displayName,
                            style = MaterialTheme.typography.titleMedium,
                            fontWeight = FontWeight.Bold,
                            color = MaterialTheme.colorScheme.onPrimaryContainer
                        )
                        if (visit.isFavorite) {
                            Icon(
                                Icons.Default.Star,
                                contentDescription = "Favorite",
                                tint = MaterialTheme.colorScheme.tertiary,
                                modifier = Modifier.size(20.dp)
                            )
                        }
                    }

                    if (visit.city != null || visit.countryCode != null) {
                        Text(
                            text = listOfNotNull(visit.city, visit.countryCode).joinToString(", "),
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onPrimaryContainer.copy(alpha = 0.7f)
                        )
                    }
                }
            }

            // Address if no POI name
            if (visit.approximateAddress != null && visit.userLabel == null && visit.poiName == null) {
                Text(
                    text = visit.approximateAddress!!,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onPrimaryContainer.copy(alpha = 0.8f),
                    modifier = Modifier.padding(top = 8.dp)
                )
            }

            // User notes
            if (visit.userNotes != null) {
                Surface(
                    modifier =
                        Modifier
                            .fillMaxWidth()
                            .padding(top = 8.dp),
                    shape = MaterialTheme.shapes.small,
                    color = MaterialTheme.colorScheme.surface.copy(alpha = 0.3f)
                ) {
                    Row(
                        modifier = Modifier.padding(8.dp),
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        Icon(
                            Icons.AutoMirrored.Filled.StickyNote2,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp),
                            tint = MaterialTheme.colorScheme.onPrimaryContainer
                        )
                        Text(
                            text = visit.userNotes!!,
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onPrimaryContainer
                        )
                    }
                }
            }

            // Metadata chips
            Row(
                modifier = Modifier.padding(top = 12.dp),
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                // Duration
                AssistChip(
                    onClick = { },
                    label = {
                        Text(
                            formatDuration(visit.duration),
                            style = MaterialTheme.typography.labelSmall
                        )
                    },
                    leadingIcon = {
                        Icon(
                            Icons.Default.Schedule,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp)
                        )
                    }
                )

                // Category
                if (visit.category != PlaceCategory.OTHER) {
                    AssistChip(
                        onClick = { },
                        label = {
                            Text(
                                visit.category.name
                                    .lowercase()
                                    .replaceFirstChar { it.uppercase() },
                                style = MaterialTheme.typography.labelSmall
                            )
                        }
                    )
                }
            }
        }
    }
}

@Composable
fun EnhancedRouteCard(
    route: com.po4yka.trailglass.domain.model.RouteSegment,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors =
            CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.secondaryContainer
            )
    ) {
        Row(
            modifier = Modifier.padding(16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Icon(
                getTransportIcon(route.transportType),
                contentDescription = null,
                tint = MaterialTheme.colorScheme.onSecondaryContainer,
                modifier = Modifier.size(28.dp)
            )

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text =
                        route.transportType.name
                            .lowercase()
                            .replaceFirstChar { it.uppercase() },
                    style = MaterialTheme.typography.titleSmall,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSecondaryContainer
                )

                Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                    Text(
                        text = "${(route.distanceMeters / 1000).toInt()} km",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSecondaryContainer
                    )

                    val duration = route.endTime - route.startTime
                    if (duration.inWholeMinutes > 0) {
                        Text(
                            text = "• ${formatDuration(duration)}",
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSecondaryContainer
                        )
                    }
                }
            }

            // Confidence indicator
            if (route.confidence < 0.7f) {
                Icon(
                    Icons.AutoMirrored.Filled.HelpOutline,
                    contentDescription = "Low confidence",
                    tint = MaterialTheme.colorScheme.onSecondaryContainer.copy(alpha = 0.5f),
                    modifier = Modifier.size(20.dp)
                )
            }
        }
    }
}

@Composable
fun DaySummaryCard(
    summary: GetTimelineUseCase.TimelineItemUI.DaySummaryUI,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors =
            CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.tertiaryContainer
            )
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Icon(
                    Icons.Default.CalendarToday,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onTertiaryContainer
                )
                Text(
                    text = summary.date.toString(),
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onTertiaryContainer
                )
            }

            Spacer(modifier = Modifier.height(8.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly
            ) {
                SummaryStatItem(
                    icon = Icons.Default.Place,
                    label = "Places",
                    value = "${summary.totalVisits}"
                )
                SummaryStatItem(
                    icon = Icons.Default.Straighten,
                    label = "Distance",
                    value = "${(summary.totalDistanceMeters / 1000).toInt()} km"
                )
                SummaryStatItem(
                    icon = Icons.AutoMirrored.Filled.DirectionsWalk,
                    label = "Routes",
                    value = "${summary.totalRoutes}"
                )
            }
        }
    }
}

@Composable
fun WeekSummaryCard(
    summary: GetTimelineUseCase.TimelineItemUI.WeekSummaryUI,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors =
            CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.tertiaryContainer
            )
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Icon(
                    Icons.Default.DateRange,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onTertiaryContainer
                )
                Column {
                    Text(
                        text = "Week Summary",
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onTertiaryContainer
                    )
                    Text(
                        text = "${summary.weekStart} - ${summary.weekEnd}",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onTertiaryContainer.copy(alpha = 0.7f)
                    )
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly
            ) {
                SummaryStatItem(
                    icon = Icons.Default.Place,
                    label = "Places",
                    value = "${summary.totalVisits}"
                )
                SummaryStatItem(
                    icon = Icons.Default.Straighten,
                    label = "Distance",
                    value = "${(summary.totalDistanceMeters / 1000).toInt()} km"
                )
                SummaryStatItem(
                    icon = Icons.Default.CalendarToday,
                    label = "Days Active",
                    value = "${summary.activeDays}"
                )
            }
        }
    }
}

@Composable
fun MonthSummaryCard(
    summary: GetTimelineUseCase.TimelineItemUI.MonthSummaryUI,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors =
            CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.tertiaryContainer
            )
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Icon(
                    Icons.Default.CalendarMonth,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onTertiaryContainer
                )
                Text(
                    text = "${summary.month.name} ${summary.year}",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onTertiaryContainer
                )
            }

            Spacer(modifier = Modifier.height(12.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly
            ) {
                SummaryStatItem(
                    icon = Icons.Default.Place,
                    label = "Places",
                    value = "${summary.totalVisits}"
                )
                SummaryStatItem(
                    icon = Icons.Default.Straighten,
                    label = "Distance",
                    value = "${(summary.totalDistanceMeters / 1000).toInt()} km"
                )
                SummaryStatItem(
                    icon = Icons.Default.DateRange,
                    label = "Weeks",
                    value = "${summary.activeWeeks}"
                )
            }

            if (summary.topCategories.isNotEmpty()) {
                Spacer(modifier = Modifier.height(8.dp))
                HorizontalDivider()
                Spacer(modifier = Modifier.height(8.dp))

                TopCategoriesSection(categories = summary.topCategories)
            }
        }
    }
}

@Composable
private fun TopCategoriesSection(
    categories: List<PlaceCategory>,
    modifier: Modifier = Modifier
) {
    Column(modifier = modifier) {
        Text(
            "Top Categories",
            style = MaterialTheme.typography.labelMedium,
            color = MaterialTheme.colorScheme.onTertiaryContainer
        )
        Spacer(modifier = Modifier.height(4.dp))
        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            categories.take(3).forEach { category ->
                AssistChip(
                    onClick = { },
                    label = {
                        Text(
                            category.name.lowercase().replaceFirstChar { it.uppercase() },
                            style = MaterialTheme.typography.labelSmall
                        )
                    },
                    leadingIcon = {
                        Icon(
                            getCategoryIcon(category),
                            contentDescription = null,
                            modifier = Modifier.size(14.dp)
                        )
                    }
                )
            }
        }
    }
}

@Composable
private fun SummaryStatItem(
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    label: String,
    value: String
) {
    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Icon(
            icon,
            contentDescription = null,
            modifier = Modifier.size(24.dp),
            tint = MaterialTheme.colorScheme.primary
        )
        Spacer(modifier = Modifier.height(4.dp))
        Text(
            text = value,
            style = MaterialTheme.typography.titleSmall,
            fontWeight = FontWeight.Bold
        )
        Text(
            text = label,
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
    }
}

File: iosApp/iosApp/Screens/Timeline/TimelineCards.swift
import SwiftUI
import Shared

/**
 * Day marker card.
 */
struct DayMarkerCard: View {
    let text: String
    let icon: String

    var body: some View {
        HStack(spacing: 8) {
            Image(systemName: icon)
                .foregroundColor(.adaptivePrimary)
            Text(text)
                .font(.body)
                .fontWeight(.medium)
        }
        .padding(12)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color(.systemGray6))
        .cornerRadius(8)
    }
}

/**
 * Enhanced visit card.
 */
struct EnhancedVisitCard: View {
    let visit: PlaceVisit

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            // Header
            HStack(alignment: .top, spacing: 12) {
                Image(systemName: categoryIcon(visit.category as? PlaceCategory ?? .other))
                    .font(.title2)
                    .foregroundColor(.blue)

                VStack(alignment: .leading, spacing: 4) {
                    HStack(spacing: 8) {
                        Text(visit.displayName)
                            .font(.headline)

                        if visit.isFavorite {
                            Image(systemName: "star.fill")
                                .foregroundColor(.yellow)
                                .font(.caption)
                        }
                    }

                    if let city = visit.city {
                        Text(city)
                            .font(.caption)
                            .foregroundColor(.secondary)
                    }
                }

                Spacer()
            }

            // Address (if no POI name)
            if let address = visit.approximateAddress, visit.userLabel == nil, visit.poiName == nil {
                Text(address)
                    .font(.caption)
                    .foregroundColor(.secondary)
            }

            // User notes
            if let notes = visit.userNotes {
                HStack(spacing: 8) {
                    Image(systemName: "note.text")
                        .font(.caption)
                    Text(notes)
                        .font(.caption)
                }
                .padding(8)
                .frame(maxWidth: .infinity, alignment: .leading)
                .background(Color(.systemGray6))
                .cornerRadius(4)
            }

            // Metadata chips
            HStack(spacing: 8) {
                HStack(spacing: 4) {
                    Image(systemName: "clock")
                        .font(.caption)
                    Text(formatDuration(visit.duration))
                        .font(.caption)
                }
                .padding(.horizontal, 8)
                .padding(.vertical, 4)
                .background(Color(.systemGray5))
                .cornerRadius(4)

                if visit.category.name != "OTHER" {
                    Text(categoryName(visit.category as? PlaceCategory ?? .other))
                        .font(.caption)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 4)
                        .background(Color(.systemGray5))
                        .cornerRadius(4)
                }
            }
        }
        .padding(16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color.blue.opacity(0.1))
        .cornerRadius(12)
    }
}

/**
 * Enhanced route card.
 */
struct EnhancedRouteCard: View {
    let route: RouteSegment

    var body: some View {
        HStack(spacing: 12) {
            Image(systemName: transportIcon(route.transportType))
                .font(.title2)
                .foregroundColor(.green)

            VStack(alignment: .leading, spacing: 4) {
                Text(transportName(route.transportType))
                    .font(.headline)

                HStack(spacing: 8) {
                    Text("\(Int(route.distanceMeters / 1000)) km")
                        .font(.caption)

                    let duration = Double(route.endTime.epochSeconds) - Double(route.startTime.epochSeconds)
                    if duration > 0 {
                        Text("• \(Int(duration / 60)) min")
                            .font(.caption)
                    }
                }
                .foregroundColor(.secondary)
            }

            Spacer()

            // Confidence indicator
            if route.confidence < 0.7 {
                Image(systemName: "questionmark.circle")
                    .foregroundColor(.secondary)
            }
        }
        .padding(16)
        .background(Color.green.opacity(0.1))
        .cornerRadius(12)
    }
}

/**
 * Day summary card.
 */
struct DaySummaryCard: View {
    let summary: GetTimelineUseCaseTimelineItemUIDaySummaryUI

    var body: some View {
        SummaryGlassCard(
            title: formatLocalDate(summary.date),
            subtitle: nil,
            icon: "calendar",
            stats: [
                (icon: "mappin.and.ellipse", label: "Places", value: "\(summary.totalVisits)"),
                (icon: "ruler", label: "Distance", value: "\(Int(summary.totalDistanceMeters / 1000)) km"),
                (icon: "figure.walk", label: "Routes", value: "\(summary.totalRoutes)")
            ]
        )
    }
}

/**
 * Week summary card.
 */
struct WeekSummaryCard: View {
    let summary: GetTimelineUseCaseTimelineItemUIWeekSummaryUI

    var body: some View {
        SummaryGlassCard(
            title: "Week Summary",
            subtitle: "\(formatLocalDate(summary.weekStart)) - \(formatLocalDate(summary.weekEnd))",
            icon: "calendar.badge.clock",
            stats: [
                (icon: "mappin.and.ellipse", label: "Places", value: "\(summary.totalVisits)"),
                (icon: "ruler", label: "Distance", value: "\(Int(summary.totalDistanceMeters / 1000)) km"),
                (icon: "calendar", label: "Days Active", value: "\(summary.activeDays)")
            ]
        )
    }
}

/**
 * Month summary card.
 */
struct MonthSummaryCard: View {
    let summary: GetTimelineUseCaseTimelineItemUIMonthSummaryUI

    var body: some View {
        VStack(spacing: 12) {
            SummaryGlassCard(
                title: "\(summary.month.name) \(summary.year)",
                subtitle: nil,
                icon: "calendar.circle",
                stats: [
                    (icon: "mappin.and.ellipse", label: "Places", value: "\(summary.totalVisits)"),
                    (icon: "ruler", label: "Distance", value: "\(Int(summary.totalDistanceMeters / 1000)) km"),
                    (icon: "calendar.badge.clock", label: "Weeks", value: "\(summary.activeWeeks)")
                ]
            )

            if !summary.topCategories.isEmpty {
                HStack(spacing: 8) {
                    ForEach(summary.topCategories.prefix(3), id: \.self) { category in
                        GlassFilterChip(
                            label: categoryName(category as? PlaceCategory ?? .other),
                            icon: categoryIcon(category as? PlaceCategory ?? .other),
                            isSelected: false,
                            tint: .blueSlate
                        ) {}
                    }
                }
            }
        }
    }
}

/**
 * Summary stat item.
 */
struct SummaryStatItem: View {
    let icon: String
    let label: String
    let value: String

    var body: some View {
        VStack(spacing: 4) {
            Image(systemName: icon)
                .font(.title3)
                .foregroundColor(.blue)
            Text(value)
                .font(.headline)
            Text(label)
                .font(.caption)
                .foregroundColor(.secondary)
        }
    }
}
